iam/src/providers/communitystaking.ts

```diff
@@ -0,0 +1,201 @@
+          total
+          round {
+            id
+          }
+        }
+      }
+    }
+      `,
+  });
+
+  const r = result as DataResult;
+
+  const response: StakeResponse = {
+    address: address,
+    totalAmountStaked: 0,
+  };
+  // Array of community stakes on the user
+  const xstake = r?.data?.data?.users[0]?.xstakeAggregates[0]?.total;
+  if (!xstake) {
+    return response;
+  }
+  const stakeAmountFormatted: string = utils.formatUnits(xstake, 18);
+  return {
+    totalAmountStaked: parseFloat(stakeAmountFormatted),
+    address: address,
+  };
+}
+
+// Export a Community Staking Bronze Stamp provider
+// User's community stake must be greater than 10 GTC
+export class CommunityStakingBronzeProvider implements Provider {
+  // Give the provider a type so that we can select it with a payload
+  type = "CommunityStakingBronze";
+  // Options can be set here and/or via the constructor
+  _options = {};
+
+  // construct the provider instance with supplied options
+  constructor(options: ProviderOptions = {}) {
+    this._options = { ...this._options, ...options };
+  }
+
+  // verify that the proof object contains valid === "true"
+  async verify(payload: RequestPayload): Promise<VerifiedPayload> {
+    let valid = false;
+    try {
+      const stakeData = await verifyStake(payload);
+      const stakeAmount = stakeData.totalAmountStaked;
+      valid = stakeAmount > 10.0;
+
+      return {
+        valid,
+        record: valid
+          ? {
+              address: stakeData.address,
+              // csgt10 = Community Staking Greater than 10
+              stakeAmount: "csgt10",
+            }
+          : {},
+      };
+    } catch (e) {
+      return {
+        valid: false,
+        error: [JSON.stringify(e)],
+      };
+    }
+  }
+}
+
+// Export a Community Staking Silver Stamp provider
+// User's community stake must be greater than 100 GTC
+export class CommunityStakingSilverProvider implements Provider {
+  // Give the provider a type so that we can select it with a payload
+  type = "CommunityStakingSilver";
+  // Options can be set here and/or via the constructor
+  _options = {};
+
+  // construct the provider instance with supplied options
+  constructor(options: ProviderOptions = {}) {
+    this._options = { ...this._options, ...options };
+  }
+
+  // verify that the proof object contains valid === "true"
+  async verify(payload: RequestPayload): Promise<VerifiedPayload> {
+    let valid = false;
+    try {
+      const stakeData = await verifyStake(payload);
+      const stakeAmount = stakeData.totalAmountStaked;
+
+      valid = stakeAmount > 100.0;
+
+      return {
+        valid,
+        record: valid
+          ? {
+              address: stakeData.address,
+              // csgt100 = Community Staking Greater than 100
+              stakeAmount: "csgt100",
+            }
+          : {},
+      };
+    } catch (e) {
+      return {
+        valid: false,
+        error: [JSON.stringify(e)],
```

Comment:
As a general rule, I would propose not to simply log or return exceptions.
In this particular example it is not critical.

But in case this would be an axios exception, and if the request we made would contain a secret in the request headers, then this might result in being leaked ...